include/dd/Edge.hpp

```diff
@@ -50,13 +50,14 @@ namespace dd {
             return reinterpret_cast<Node*>(reinterpret_cast<std::uintptr_t>(p) | 4ULL);
         }
 
-        //        static inline void setDensityConjugateEdgeTrue(Edge *e) {
-        //            e->p = reinterpret_cast<Node*>(reinterpret_cast<std::uintptr_t>(e->p) | 1ULL);
-        //        }
-        //
-        //        static inline void setDensityMatrixEdgeTrue(Edge *e) {
-        //            e->p = reinterpret_cast<Node*>(reinterpret_cast<std::uintptr_t>(e->p) | 4ULL);
-        //        }
+        [[nodiscard]] static inline bool tempDensityMatrixFlagsEqual(const std::uint8_t a, const std::uint8_t b) {
+            return (a & (7U)) == (b & (7U));
+        }
+
+        [[nodiscard]] static inline bool returnTempDensityMatrixFlags(const std::uint8_t a) {
+            return a & (7U) ;
+        }
+
```

Comment:
these functions should not be here as they do not belong in the `Edge` class. 
both of these are functions on nodes, more precisely, on a node's flags.
Along the lines of:
```c++
        [[nodiscard]] inline bool getTempDensityMatrixFlags() const {
            return flags & (7U);
        }
        
        [[nodiscard]] static inline bool tempDensityMatrixFlagsEqual(const Node& a, const Node& b) {
            return a.getTempDensityMatrixFlags() == b.getTempDensityMatrixFlags();
        }
```